util/rlp/src/stream.rs

```diff
@@ -52,17 +53,74 @@ impl RlpStream {
 	}
 
 	/// Initializes the `Stream` as a list.
-	pub fn new_list(len: usize) -> Self {
+	fn new_list(len: usize) -> Self {
 		let mut stream = RlpStream::new();
 		stream.begin_list(len);
 		stream
 	}
 
+	/// Apends null to the end of stream, chainable.
+	///
+	/// ```rust
+	/// extern crate rlp;
+	/// extern crate stream_encoder;
+	/// use rlp::*;
+	/// use stream_encoder::Stream;
+	///
+	/// fn main () {
+	/// 	let mut stream = RlpStream::new_list(2);
+	/// 	stream.append_empty_data().append_empty_data();
+	/// 	let out = stream.out();
+	/// 	assert_eq!(out, vec![0xc2, 0x80, 0x80]);
+	/// }
+	/// ```
+	fn append_empty_data(&mut self) -> &mut Self {
+		// self push raw item
+		self.buffer.push(0x80);
+
+		// try to finish and prepend the length
+		self.note_appended(1);
+
+		// return chainable self
+		self
+	}
+
+	/// Drain the object and return the underlying ElasticArray. Panics if it is not finished.
+	fn drain(self) -> ElasticArray1024<u8> {
```

Comment:
What are the reasons for using `ElasticArray` and not `smallvec`? Are there any subtle differences?

Reply:
I know @Vurich optimized smallvec somewhat recently, but there were noticable performance regressions when we tried to switch last year.

Reply:
@ordian No reason other than sticking to what the previous code did trying to keep the PR on scope. :)
I am a little worried about the slight but consistent performance regression of this PR so I'm hesitant to toss `smallvec` into the mix at this point, especially after reading about the outcome of last years attempt ([here]https://github.com/paritytech/parity/pull/4874)).